src/agents/toolkits/site-spec.js

```diff
@@ -11,82 +11,70 @@ import {
 } from '../tools/site-tools.js';
 import uuidv4 from '../../utils/uuid.js';
 
-const INITIAL_STATE = {
-	textColor: undefined,
-	backgroundColor: undefined,
-	accentColor: undefined,
-	siteTitle: undefined,
-	siteDescription: undefined,
-	siteTopic: undefined,
-	siteType: undefined,
-	siteLocation: undefined,
-	pages: [],
-};
-
 class SimpleSiteToolkit extends Toolkit {
-	constructor( props ) {
-		super( props, INITIAL_STATE );
+	constructor( props, stateManager ) {
+		super( props, stateManager );
 
 		this.tools = this.getTools();
 	}
 
 	setSiteColors = ( { textColor, backgroundColor, accentColor } ) => {
 		if ( textColor ) {
-			this.setState( { textColor } );
+			this.stateManager.setState( { textColor } );
 		}
 		if ( backgroundColor ) {
-			this.setState( { backgroundColor } );
+			this.stateManager.setState( { backgroundColor } );
 		}
 		if ( accentColor ) {
-			this.setState( { accentColor } );
+			this.stateManager.setState( { accentColor } );
 		}
 		return 'Site colors updated';
 	};
 
 	setSiteTitle = ( { value } ) => {
-		this.setState( { siteTitle: value } );
+		this.stateManager.setState( { siteTitle: value } );
 		return `Site title set to "${ value }"`;
 	};
 
 	setSiteDescription = ( { value } ) => {
-		this.setState( { siteDescription: value } );
+		this.stateManager.setState( { siteDescription: value } );
```

Comment:
IMO this pattern is adding of visual noise for little benefit other than exposing the inner workings of the ToolKit class. Why not just have the `toolkit.setState()` method delegate to `toolkit.stateManager.setState()`, instead of referring to it directly?